### PR TITLE
Add Storybook components 

### PR DESCRIPTION
**Pull request by SebastianStupp**
https://loginicer-pi-storybook-oitrzc2.herokuapp.com/storybook/index.html?path=/story/input--example-input

Screenshot 2020-04-12 at 17 52 39
Screenshot 2020-04-12 at 17 52 20

### CODE REVIEW

File: client/src/components/AppContainer.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import styled from '@emotion/styled';
+
+const AppContainer = styled.div`
+  width: 100vw;
```
Comment (florianGierlichs) [inline client/src/components/AppContainer.js:4]: I assume, you will use this container not only in your App, so I would recommend a different name. How about FlexColumnContainer?

File: client/src/components/Menu.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+export default function DefaultMenu() {
+  return (
+    <>
+      <AppContainer>
```
Comment (florianGierlichs) [inline client/src/components/Menu.js:18]: This should be named MenuContainer or something more common.

File: client/src/stories/AppContainer.stories.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import AppContainer from '../components/AppContainer';
+
+export default {
+  title: 'AppContainer',
```
Comment (florianGierlichs) [inline client/src/stories/AppContainer.stories.js:5]: AppContainer is only a flexbox, so this shouldn't show anything does it?

**Comment by florianGierlichs** [comment]
This is not a good pull request in my opinion. You have 29 changed files, that's way too many changes for this topic. Also, it makes it much more difficult to review and get a good look at the changes.
You really should structure your pull-requests more detailed. Also, what does "storybook" as a topic mean? The title should indicate what you have done. How about "Add storybook components"?

**Comment by marwinburesch** [comment]
I agree with @florianGierlichs 
setting up storybook should be one PR and then you can have separate PRs for each component where you see how they function in storybook and then implement them for the app once everything works as expected

File: client/src/components/Dropdown.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export default function DefaultDropdown() {
+  const [dropdownValue, setdropdownValue] = React.useState('');
+
+  const handleChange = (event) => {
+    setdropdownValue(event.target.value);
+  };
+
+  return (
+    <Dropdown value={dropdownValue} onChange={handleChange}>
+      <option value="" selected disabled hidden>
```
Comment (marwinburesch) [inline client/src/components/Dropdown.js:20]: why `hidden` ?

File: client/src/components/GlobalStyles.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <Global
+      styles={(theme) => css`
+        @import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');
+        *,
+        *:before,
+        *:after {
+          box-sizing: border-box;
+        }
+        body {
+          height: 100vh;
```
Comment (marwinburesch) [inline client/src/components/GlobalStyles.js:15]: you App container already has `100vh` I don't think you need it

File: client/src/components/AppContainer.js (hunk 1 above)
Comment (SebastianStupp) [reply client/src/components/AppContainer.js:4]: Yeah seems like I don't need the Container anyway, so I removed it:)

File: client/src/components/Menu.js (hunk 2 above)
Comment (SebastianStupp) [reply client/src/components/Menu.js:18]: Yeah seems like I don't need the Container anyway, so I removed it:)

File: client/src/stories/AppContainer.stories.js (hunk 3 above)
Comment (SebastianStupp) [reply client/src/stories/AppContainer.stories.js:5]: Yeah seems like I don't need the Container anyway, so I removed it:)

File: client/src/components/MenuButtons.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+export const MenuButtonTransparent = styled.button`
+  background-color: rgb(219, 143, 0, 63%);
+  width: 100%;
+  max-width: 500px;
+  height: 30%;
+  font-size: 2rem;
+  color: #0d3955;
+  padding: 0px;
+  border: 0;
+`;
```
Comment (marwinburesch) [inline client/src/components/MenuButtons.js:25]: you could just re-style `MenuButton` or design it in a way that changes some values according to what props it receives

File: client/src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import MenuButton from '../assets/menubutton.svg';
+import PropTypes from 'prop-types';
+
+const Header = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-around;
+  margin-top: 10px;
+  margin-bottom: 20px;
+  top: 0;
```
Comment (marwinburesch) [inline client/src/components/Header.js:14]: what is this for?

File: client/.storybook/themeDecorator.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { ThemeProvider } from 'emotion-theming';
+import GlobalStyles from '../src/components/GlobalStyles';
+import theme from '../src/theme';
+
+const ThemeDecorator = (storyFn) => (
+  <ThemeProvider theme={theme}>
+    <GlobalStyles />
```
Comment (marwinburesch) [inline client/.storybook/themeDecorator.js:8]: this can be added in the storybook config. Doesn't really belong in the `ThemeProvider`

File: client/package.json [hunk 9]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "emotion-theming": "^10.0.27",
```
Comment (marwinburesch) [inline client/package.json:11]: should've been part of the setup or a separate PR to add theming/themes

File: client/src/components/Dropdown.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import React from 'react';
+
+const Dropdown = styled.select`
+  background-color: ${(props) => props.theme.colors.secondary};
+  border-radius: 3px;
+  min-width: 254px;
+  min-height: 34px;
+`;
+
+export default function DefaultDropdown() {
```
Comment (marwinburesch) [inline client/src/components/Dropdown.js:11]: This component should receive an array and then map it as `option `elements

File: client/src/components/Menu.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import Header from './Header';
+import AppContainer from './AppContainer';
+import { MenuButton, MenuButtonTransparent } from './MenuButtons';
+
+const MenuContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
```
Comment (marwinburesch) [inline client/src/components/Menu.js:12]:
```suggestion
  align-items: center;
  & > button {
  background-color: rgb(219, 143, 0);
  }
  & :nth-child(3n) {
  background-color: rgb(219, 143, 0, 63%);
  }
```

this could also give your `MenuButton` components different backgrounds

File: client/src/components/Menu.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <>
+      <AppContainer>
+        <Header type="logout"></Header>
+        <MenuContainer>
+          <MenuButton>Client Master</MenuButton>
+          <MenuButtonTransparent>Item Master</MenuButtonTransparent>
+          <MenuButton>Storage System</MenuButton>
+          <MenuButtonTransparent>Inbound</MenuButtonTransparent>
+          <MenuButton>Outbound</MenuButton>
```
Comment (marwinburesch) [inline client/src/components/Menu.js:25]: just `map()` these guys from an array

File: client/src/theme.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+    basic: 'linear-gradient(#0d3955 60%, black);',
+    textprimary: 'linear-gradient(#0d3955 60%, black)',
+    textsecondary: '#fff6d6',
+    primary: '#db8f00',
+    secondary: '#fff6d6',
+    tertiary: '#707070',
+  },
+};
+
+export default theme;
```
Comment (marwinburesch) [inline client/src/theme.js:12]: all this could be a separate PR that adds `emotion/theming` as well

File: package.json [hunk 14]
```diff
@@ -44,7 +44,8 @@
   "dependencies": {
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
-    "mongodb": "^3.5.5"
+    "mongodb": "^3.5.5",
+    "prop-types": "^15.7.2"
```
Comment (marwinburesch) [inline package.json:48]: should be separate as well, and used :P
